scripts/functions/checksum

```diff
@@ -195,14 +195,24 @@ __rvm_checksum_read()
 
   for _name in "${list[@]}"
   do
+    # md5
     _checksum_md5="$(      __rvm_db_ "$rvm_path/config/md5"    "$_name" | \command \head -n 1 )"
+
     [[ -n "${_checksum_md5:-}" ]] ||
       _checksum_md5="$(    __rvm_db_ "$rvm_user_path/md5"      "$_name" | \command \head -n 1 )"
 
+    [[ -n "${_checksum_md5:-}" && $_name == http* ]] ||
```

Comment:
This is a bug, specifically it will use `curl` when `-n "${_checksum_md5:-}"`, that is when `_checksum_md5` is not empty, i.e, when it's already set from config files, and when `$_name` does not start with `http`. Fix in https://github.com/rvm/rvm/pull/4707